frontend/pulse3d/src/components/basicWidgets/UsageProgressWidget.js

```diff
@@ -20,52 +20,70 @@ const ExpiredP = styled.p`
 `;
 
 export default function UsageProgressWidget({ colorOfTextLabel }) {
-  const { usageQuota } = useContext(AuthContext);
-  const [maxUploads, setMaxUploads] = useState(-1);
-  const [actualUploads, setActualUploads] = useState(0);
+  const { usageQuota, setUsageQuota } = useContext(AuthContext);
+  const [maxAnalyses, setMaxAnalyses] = useState();
+  const [actualAnalyses, setActualAnalyses] = useState(0);
   const [usagePercentage, setUsagePercentage] = useState(0);
-
+  const [isExpired, setIsExpired] = useState();
+  const pollUsageQuota = async () => {
+    try {
+      const response = await fetch(`${process.env.NEXT_PUBLIC_PULSE3D_URL}/usage?service=pulse3d`);
+      if (response && response.status === 200) {
+        const newUsageQuota = await response.json();
+        const limit = parseInt(newUsageQuota.limits.jobs);
+        const actual = parseInt(newUsageQuota.current.jobs);
+        const newUsagePercentage = Math.round((actual / limit) * 100);
+        if (newUsagePercentage > 100) {
+          setUsagePercentage(100);
+        } else {
+          setUsagePercentage(newUsagePercentage);
+        }
+        setActualAnalyses(newUsageQuota["current"]["jobs"]);
+        setIsExpired(newUsageQuota.jobs_reached);
+        setUsageQuota(newUsageQuota);
+      }
+    } catch (e) {
+      console.log("ERROR fetching usage quota in /usage");
+    }
+  };
   useEffect(() => {
     if (usageQuota && usageQuota.limits && usageQuota.current) {
       const limit = parseInt(usageQuota.limits.jobs);
       const actual = parseInt(usageQuota.current.jobs);
-      setMaxUploads(limit);
-      setActualUploads(actual);
-      const usagePercentage = parseInt((actual / limit) * 100);
+      setMaxAnalyses(limit);
+      setActualAnalyses(actual);
+      const usagePercentage = Math.round((actual / limit) * 100);
       if (usagePercentage > 100) {
         setUsagePercentage(100);
       } else {
         setUsagePercentage(usagePercentage);
       }
+      setIsExpired(usageQuota.jobs_reached);
     }
+    pollUsageQuota();
   }, [usageQuota]);
 
-  const CorrectCardComponent = () => {
-    let usageState;
-    if (maxUploads === -1) {
-      // Unlimited mode
-      usageState = (
-        <>
-          <ProgressDiv>Unlimited Access</ProgressDiv>
-        </>
-      );
-    } else if (!usageQuota.jobs_reached) {
-      // max usage has not been reached
-      usageState = (
+  useEffect(() => {
+    if (actualAnalyses) {
```

Comment:
is a new interval going to be started every time the actualAnalyses value changes?

Reply:
I can confirm that the set interval is not being duplicated

Reply:
anytime the interval is started the previous one is cancelled